cumulus/pallets/parachain-system/src/validate_block/implementation.rs

```diff
@@ -48,6 +50,9 @@ fn with_externalities<F: FnOnce(&mut dyn Externalities) -> R, R>(f: F) -> R {
 	sp_externalities::with_externalities(f).expect("Environmental externalities not set.")
 }
 
+/// Recorder instance to be used during this validate_block call.
+environmental::environmental!(recorder: trait ProofSizeProvider);
```

Comment:
Why not having the proof size recorder accessed from externalities (I mean it seems to be use in the trie backend, thus probably accessible) ?

Reply:
It is not available from the externalities. We only have access to what is exposed by the [`Externalities` ](https://github.com/skunert/polkadot-sdk/blob/44a76cd322f5c843856cb4ecb15caa237cf9b25e/substrate/primitives/externalities/src/lib.rs#L82) trait. The recorder here lives in an extension, which are currently not available for no-std. I do not see a blocker why we could not make them available for no-std, but not sure this is required since this location here is the only use case.